core/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java

```diff
@@ -739,12 +684,59 @@ public void testMalformedQuery() throws IOException {
                 "        }\n" +
                 "    }\n" +
                 "}";
+        expectParsingException(json, "[query] is already defined.");
+    }
 
-        try {
-            parseQuery(json);
-            fail("parse should have failed");
-        } catch(ParsingException e) {
-            assertThat(e.getMessage(), containsString("[query] is already defined."));
+    private void expectParsingException(String json, Matcher<String> messageMatcher) {
+        ParsingException e = expectThrows(ParsingException.class, () -> parseQuery(json));
+        assertThat(e.getMessage(), messageMatcher);
+    }
+
+    private void expectParsingException(String json, String message) {
+        expectParsingException(json, equalTo("failed to parse [function_score] query. " + message));
+    }
+
+    /**
+     * A hack on top of the normal random score function that fixed toQuery to work properly in this unit testing environment.
+     */
+    public static class RandomScoreFunctionBuilderWithFixedSeed extends RandomScoreFunctionBuilder {
+        public static final String[] NAMES = new String[] {"random_with_fixed_seed"};
+
+        public RandomScoreFunctionBuilderWithFixedSeed() {
+        }
+
+        /**
+         * Read from a stream.
+         */
+        public RandomScoreFunctionBuilderWithFixedSeed(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        protected int getCurrentShardId() {
```

Comment:
is it an option to make this method package private? Then it would become more of an internal thing. Thanks for addressing this!